File: backend/app/api/conversations.py
"""Conversation API routes — CRUD for conversations and messages."""

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db
from app.dependencies import get_current_user
from app.models.user import User
from app.schemas.conversation import (
    ConversationCreate,
    ConversationDetailResponse,
    ConversationListResponse,
    ConversationResponse,
    ConversationUpdate,
)
from app.services import conversation_service

router = APIRouter(prefix="/api/conversations", tags=["conversations"])


@router.get("", response_model=ConversationListResponse)
async def list_conversations(
    page: int = Query(1, ge=1),
    per_page: int = Query(50, ge=1, le=100),
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    """List the current user's conversations, newest first."""
    return await conversation_service.list_conversations(db, current_user.id, page, per_page)


@router.post("", response_model=ConversationResponse, status_code=201)
async def create_conversation(
    data: ConversationCreate,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    """Create a new empty conversation."""
    return await conversation_service.create_conversation(db, current_user.id, data)


@router.get("/{conversation_id}", response_model=ConversationDetailResponse)
async def get_conversation(
    conversation_id: int,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    """Get a conversation with all its messages."""
    return await conversation_service.get_conversation(db, conversation_id, current_user.id)


@router.put("/{conversation_id}", response_model=ConversationResponse)
async def update_conversation(
    conversation_id: int,
    data: ConversationUpdate,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    """Update a conversation's title."""
    return await conversation_service.update_conversation(db, conversation_id, current_user.id, data.title)


@router.delete("/{conversation_id}", status_code=204)
async def delete_conversation(
    conversation_id: int,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    """Delete a conversation and all its messages."""
    await conversation_service.delete_conversation(db, conversation_id, current_user.id)

File: frontend/src/api/conversations.ts
import apiClient from './client';
import type { Conversation, ConversationDetail, ConversationListResponse } from '../types/chat';

export async function listConversations(page = 1, perPage = 50): Promise<ConversationListResponse> {
  const res = await apiClient.get<ConversationListResponse>('/conversations', {
    params: { page, per_page: perPage },
  });
  return res.data;
}

export async function createConversation(title = 'New Conversation'): Promise<Conversation> {
  const res = await apiClient.post<Conversation>('/conversations', { title });
  return res.data;
}

export async function getConversation(id: number): Promise<ConversationDetail> {
  const res = await apiClient.get<ConversationDetail>(`/conversations/${id}`);
  return res.data;
}

export async function updateConversation(id: number, title: string): Promise<Conversation> {
  const res = await apiClient.put<Conversation>(`/conversations/${id}`, { title });
  return res.data;
}

export async function deleteConversation(id: number): Promise<void> {
  await apiClient.delete(`/conversations/${id}`);
}
